usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -0,0 +1,292 @@
+import logging
+import re
+
+from django.core.management import BaseCommand
+from django.db import connection, transaction
+
+from usaspending_api.common.helpers.sql_helpers import ordered_dictionary_fetcher
+
+logger = logging.getLogger("script")
+
+
+class Command(BaseCommand):
+    help = """
+    This command is used to swap two tables; the current and a new table with "_temp" appended.
+    Validation is run against the new table to ensure that after the swap is complete all of the indexes, constraints,
+    columns, and table name will be the same.
+
+    NOTE: This entire process IS NOT ATOMIC (only the final swap of tables)!
+    This choice was made to prevent potential deadlock scenarios since the swapping / renaming of indexes, constraints,
+    and tables should only take milliseconds. If this command fails then any cleanup will have to be done manually.
+    Any failure in this command should not affect current API since the final "DROP TABLE" command is not run
+    until the very end.
+    """
+
+    # Values are set in the beginning of "handle()"
+    db_table_name: str
+    temp_db_table_name: str
+
+    # Query values are populated as they are run during validation and saved for re-use
+    query_result_lookup = {
+        "temp_table_constraints": [],
+        "curr_table_constraints": [],
+        "temp_table_indexes": [],
+        "curr_table_indexes": [],
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--table",
+            type=str,
+            help="The Postgres Table to swap with another containing the same name with '_temp' appended",
+        )
+        parser.add_argument(
+            "--keep-old-data",
+            action="store_true",
+            default=False,
+            help="Indicates whether or not to drop old table at the end of the command",
+        )
+        parser.add_argument(
+            "--allow-foreign-key",
```

Comment:
maybe `--keep-foreign-keys`

I need to read on to see if this is a guard, that halts the cmd if there _are_ FKs on the target table, or if this is dicatating whether the new table gets the old's FKs or not. This is not clear from the help docs.